proposals/simd/portable-simd.md

```diff
@@ -0,0 +1,1324 @@
+* `b32x4.extractLane(a: b32x4, i: LaneIdx4) -> boolean`
+* `b64x2.extractLane(a: b64x2, i: LaneIdx2) -> boolean`
+* `i8x16.extractLane(a: v128, i: LaneIdx16) -> i8`
+* `i16x8.extractLane(a: v128, i: LaneIdx8) -> i16`
+* `i32x4.extractLane(a: v128, i: LaneIdx4) -> i32`
+* `i64x2.extractLane(a: v128, i: LaneIdx2) -> i64`
+* `f32x4.extractLane(a: v128, i: LaneIdx4) -> f32`
+* `f64x2.extractLane(a: v128, i: LaneIdx2) -> f64`
+
+Extract the value of lane `i` in `a`.
+
+```python
+def S.extractLane(a, i):
+    return a[i]
+```
+
+### Replace lane value
+* `b8x16.replaceLane(a: b8x16, i: LaneIdx16, x: boolean) -> b8x16`
+* `b16x8.replaceLane(a: b16x8, i: LaneIdx8, x: boolean) -> b16x8`
+* `b32x4.replaceLane(a: b32x4, i: LaneIdx4, x: boolean) -> b32x4`
+* `b64x2.replaceLane(a: b64x2, i: LaneIdx2, x: boolean) -> b64x2`
+* `i8x16.replaceLane(a: v128, i: LaneIdx16, x: i8) -> v128`
+* `i16x8.replaceLane(a: v128, i: LaneIdx8, x: i16) -> v128`
+* `i32x4.replaceLane(a: v128, i: LaneIdx4, x: i32) -> v128`
+* `i64x2.replaceLane(a: v128, i: LaneIdx2, x: i64) -> v128`
+* `f32x4.replaceLane(a: v128, i: LaneIdx4, x: f32) -> v128`
+* `f64x2.replaceLane(a: v128, i: LaneIdx2, x: f64) -> v128`
+
+Return a new vector with lanes identical to `a`, except for lane `i` which has
+the value `x`.
+
+```python
+def S.replaceLane(a, i, x):
+    result = S.New()
+    for j in range(S.Lanes):
+        result[j] = a[j]
+    result[i] = x
+    return result
+```
+
+### Lane-wise select
+* `v8x16.select(s: b8x16, t: v128, f: v128) -> v128`
+* `v16x8.select(s: b16x8, t: v128, f: v128) -> v128`
+* `v32x4.select(s: b32x4, t: v128, f: v128) -> v128`
+* `v64x2.select(s: b64x2, t: v128, f: v128) -> v128`
+
+Use a boolean vector to select lanes from two numerical vectors.
+
+```python
+def S.select(s, t, f):
+    result = S.New()
+    for i in range(S.Lanes):
+        if s[i]:
+            result[i] = t[i]
+        else
+            result[i] = f[i]
+    return result
+```
+
+### Swizzle lanes
+* `v8x16.swizzle(a: v128, s: LaneIdx16[16]) -> v128`
+* `v16x8.swizzle(a: v128, s: LaneIdx8[8]) -> v128`
+* `v32x4.swizzle(a: v128, s: LaneIdx4[4]) -> v128`
+* `v64x2.swizzle(a: v128, s: LaneIdx2[2]) -> v128`
+
+Create vector with lanes rearranged:
+
+```python
+def S.swizzle(a, s):
+    result = S.New()
+    for i in range(S.Lanes):
+        result[i] = a[s[i]]
+    return result
+```
+
+### Shuffle lanes
+* `v8x16.shuffle(a: v128, b: v128, s: LaneIdx32[16]) -> v128`
+* `v16x8.shuffle(a: v128, b: v128, s: LaneIdx16[8]) -> v128`
+* `v32x4.shuffle(a: v128, b: v128, s: LaneIdx8[4]) -> v128`
+* `v64x2.shuffle(a: v128, b: v128, s: LaneIdx4[2]) -> v128`
+
+Create vector with lanes selected from the lanes of two input vectors:
+
+```python
+def S.shuffle(a, b, s):
+    result = S.New()
+    for i in range(S.Lanes):
+        if s[i] < S.lanes:
+            result[i] = a[s[i]]
+        else:
+            result[i] = b[s[i] - S.lanes]
+    return result
+```
+
+## Integer arithmetic
```

Comment:
Ah apologies, I though I'd seen it in one of the files in your PR, and it was missing here. This is good then!